addons/project/views/project_task_views.xml

```diff
@@ -875,6 +875,33 @@
             </field>
         </record>
 
+        <record id="project_task_kanban_view_project_milestone" model="ir.ui.view">
+            <field name="name">project.task.kanban.inherit.project.milestone</field>
+            <field name="model">project.task</field>
+            <field name="inherit_id" ref="view_task_kanban"/>
+            <field name="mode">primary</field>
+            <field name="arch" type="xml">
+                <xpath expr="//kanban" position="attributes">
+                    <attribute name="default_group_by">milestone_id</attribute>
+                </xpath>
+            </field>
+        </record>
+
+        <record id="project_task_tree_view_project_milestone" model="ir.ui.view">
+            <field name="name">project.task.view.tree.project.milestone</field>
+            <field name="model">project.task</field>
+            <field name="inherit_id" ref="project_task_view_tree_base"/>
+            <field name="mode">primary</field>
+            <field name="arch" type="xml">
+                <xpath expr="//tree" position="attributes">
+                    <attribute name="default_group_by">milestone_id</attribute>
+                </xpath>
+                <xpath expr="//field[@name='milestone_id']" position="attributes">
+                    <attribute name="optional">show</attribute>
+                </xpath>
+            </field>
+        </record>
+
```

Comment:
I don't see how useful is showing the field `milestone_id` in list view when grouped by milestone.
As, apart from that, those inherited views only change the groupby, I would suggest to rather use the base views, and adding `'search_default_group_by_milestone_id': 1` in the context.
What do you think? :slightly_smiling_face:

Reply:
Hello @auon-odoo  I think aju prefers default group by in views instead of search view. Still, I will check with her and get back to you.

Reply:
ok thanks :slightly_smiling_face:

Reply:
Hello, @auon-odoo Aju agrees to go with my changes. Thanks

Reply:
Also Displaying milestone field default in list view is better as user normally use this view to change milestone in batch I guess :sweat_smile:

Reply:
Mh, you're right :fire: